src/Dapr.Extensions.Configuration/DaprConfigurationStoreExtension.cs

```diff
@@ -0,0 +1,94 @@
+        /// <param name="keys">The keys, if any, to request. If empty, returns all configuration items.</param>
+        /// <param name="client">The <see cref="DaprClient"/> used for the request.</param>
+        /// <param name="sidecarWaitTimeout">The <see cref="TimeSpan"/> used to configure the timeout waiting for Dapr.</param>
+        /// <param name="metadata">Optional metadata sent to the configuration store.</param>
+        /// <param name="cancellationToken"><see cref="CancellationToken"/> used to cancel the operation.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddDaprConfigurationStore(
+            this IConfigurationBuilder configurationBuilder,
+            string store,
+            IReadOnlyList<string> keys,
+            DaprClient client,
+            TimeSpan sidecarWaitTimeout,
+            IReadOnlyDictionary<string, string>? metadata = default,
+            CancellationToken cancellationToken = default)
+        {
+            ArgumentVerifier.ThrowIfNullOrEmpty(store, nameof(store));
+            ArgumentVerifier.ThrowIfNull(keys, nameof(keys));
+            ArgumentVerifier.ThrowIfNull(client, nameof(client));
+
+            configurationBuilder.Add(new DaprConfigurationStoreSource()
+            {
+                Store = store,
+                Keys = keys,
+                Client = client,
+                SidecarWaitTimeout = sidecarWaitTimeout,
+                IsStreaming = false,
+                Metadata = metadata,
+                CancellationToken = cancellationToken
+            });
+
+            return configurationBuilder;
+        }
+
+        /// <summary>
+        /// Register a streaming configuration store. This opens a stream to the Dapr Configuration API which
+        /// will get updates to keys should any occur. This stream will not be closed until canceled with the
+        /// <see cref="CancellationToken"/> or the configuration is unsubscribed in Dapr.
+        /// </summary>
+        /// <param name="configurationBuilder">The <see cref="IConfigurationBuilder"/> to add to.</param>
+        /// <param name="store">The configuration store to query.</param>
+        /// <param name="keys">The keys, if any, to request. If empty, returns all configuration items.</param>
+        /// <param name="client">The <see cref="DaprClient"/> used for the request.</param>
+        /// <param name="sidecarWaitTimeout">The <see cref="TimeSpan"/> used to configure the timeout waiting for Dapr.</param>
+        /// <param name="metadata">Optional metadata sent to the configuration store.</param>
+        /// <param name="cancellationToken"><see cref="CancellationToken"/> used to cancel the operation.</param>
+        /// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
+        public static IConfigurationBuilder AddStreamingDaprConfigurationStore(
```

Comment:
BTW I like the approach of giving this a different method name. It's very obvious what the difference is.